community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -157,6 +158,26 @@
      */
     Type transactionType();
 
+    /**
+     * Return transaction id if it was already assigned or {@link TransactionData#UNASSIGNED_TRANSACTION_ID} otherwise.
+     * Transaction id is assigned as soon as transaction committed.
+     * @return transaction id.
+     */
+    default long getTransactionId()
```

Comment:
I think that's a good package name. The kernel API is _not_ a public surface users are expected to interact with, even if they obviously _can_. Other components create our public surface.